### PR TITLE
feat: declareDirective support

### PR DESCRIPTION
**Pull request by Steven-Harris**
Add support for declareDirective in the createDirectiveFactory so that
directive modules can be tested

## PR Checklist

- [X] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #303

## What is the new behavior?
Allow for declareDirective option in the createDirectoryFactory

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,30 +30,44 @@ Spectator helps you get rid of all the boilerplate grunt work, leaving you with
 
 ## Table of Contents
 
+- [Features](#features)
```
Comment (NetanelBasal) [inline README.md:33]: Why do we need Features and Table of Contents?

**Comment by NetanelBasal** [comment]
Add yourself as a contributor. `yarn contributor:add`

File: README.md (hunk 1 above)
Comment (Steven-Harris) [reply README.md:33]: hmm nice catch, I have no idea why this changed. I will remove it

Comment (Steven-Harris) [reply README.md:33]: ahh it was my previewer trying to be "helpful"

**Comment by Steven-Harris** [comment]
Let me know if there is anything else

**Pull request merged**